shlr/gdb/src/gdbclient/core.c

```diff
@@ -743,7 +743,7 @@ int gdbr_write_register(libgdbr_t *g, int index, char *value, int len) {
 		return -1;
 	}
 	reg_cache.valid = false;
-	ret = snprintf (command, sizeof (command) - 1, "%s%d=", CMD_WRITEREG, index);
+	ret = snprintf (command, sizeof (command) - 1, "%s%x=", CMD_WRITEREG, index);
```

Comment:
Scary that it ever worked at all

Reply:
Yup, it worked for the first 10 registers. I was testing with `r5` at first (arbitrary), and using `dr` to print the registers before and after a change. The length of the list had me scrolling up every time to see the new value, so I switched to `r10`, and suddenly it stopped working - took me a while to understand why :)